spec/registry.dtd

```diff
@@ -0,0 +1,33 @@
+<!ELEMENT registry (function*|pattern*)>
+
+<!ELEMENT function (description|signature+)>
+<!ATTLIST function name NMTOKEN #REQUIRED>
+
+<!ELEMENT description (#PCDATA)>
+
+<!ELEMENT pattern EMPTY>
+<!ATTLIST pattern id ID #REQUIRED>
+<!ATTLIST pattern regex CDATA #REQUIRED>
+
+<!ELEMENT signature (input*|param*|match*)>
+<!ATTLIST signature type (match|format) #REQUIRED>
+<!ATTLIST signature position (open|close|standalone) "standalone">
+<!ATTLIST signature locales NMTOKENS #IMPLIED>
+
+<!ELEMENT input EMPTY>
+<!ATTLIST input title CDATA #IMPLIED>
+<!ATTLIST input values NMTOKENS #IMPLIED>
```

Comment:
If we end up allowing for some `nmtoken`-ish values to be used unquoted as arguments (#364), it would be good to match that rule here.

Reply:
I don't think #364 is related, actually. My intent here was to allow specifying either an enumeration of nmtokens or a regex to validate arguments that are MF2 literals. This seems orthogonal to whether these literals are quoted or not?

Reply:
I meant that if #364 lands, then _nearly_ all `NMTOKEN` values may be used without quoting, except for the ones that start with `:` and `-`. Given that, it might be appropriate to leave those out of the values that are definable by `values` rather than `pattern`.

Reply:
I don't think we can enforce such restriction in the DTD alone. LDML does it by extending DTD with _annotations_: https://unicode.org/reports/tr35/#57-dtd-annotations.